p2p/src/serv.rs

```diff
@@ -87,10 +87,9 @@ impl Server {
 						let sc = stream.try_clone();
 						if let Err(e) = self.handle_new_peer(stream) {
 							warn!("Error accepting peer {}: {:?}", peer_addr.to_string(), e);
-						} else {
-							if let Ok(s) = sc {
-								connected_sockets.insert(peer_addr, s);
-							}
+							let _ = self.peers.update_state(peer_addr, State::Banned);
```

Comment:
👍 the idea to ban the handshake failed peer should be the killer for this annoying connection exhaustion problem!

but a quick question here, I think we don't add this peer address into the peers list and database if handshake fail, so the `update_state` here should be a `save_peer` call?

Reply:
You're right, I thought `update_state` would still save the peer address with the state (as other other pieces of data are unknown) but looks like it'll fail instead. Will fix.

Reply:
.